### PR TITLE
feat: Share forms publicly and collect submissions with Endatix Hub

### LINKED ISSUE #122: [FEATURE] Share forms publicly and collect submissions with Endatix Hub
## Overview
Introduce a public page for sharing forms so that they can be distributed to collect submissions. The design should follows the [wireframes](https://www.figma.com/design/2bwMXMuvAOJKYXemu5HjhX/Endatix-Hub---Wireframes?node-id=0-1&node-type=canvas&t=wNT5Pb2boChRN0gM-0)

## Details
Forms should be accessible by potential respondents through a publicly accessible URL

## Proposed Solution
Create a route for accessing forms by formId, which does not require authentication.

## Challenges/Risks
None

## Deliverables
One new page

## Resources
- [SurveyJS' Next.JS client sample](https://github.com/surveyjs/surveyjs-nextjs/blob/main/src/app/survey/page.tsx)
- [Next.JS routing docs](https://nextjs.org/docs/app/building-your-application/routing/dynamic-routes)

### PR DESCRIPTION
**Pull request by orilux**
# Share forms publicly and collect submissions with Endatix Hub

## Description
Introduces a public page for sharing forms so that they can be distributed to collect submissions. The design follows the [wireframes](https://www.figma.com/design/2bwMXMuvAOJKYXemu5HjhX/Endatix-Hub---Wireframes?node-id=0-1&node-type=canvas&t=wNT5Pb2boChRN0gM-0)

## Related Issues
Closes #122.

## Type of Change
Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Screenshots
not needed

### CODE REVIEW

File: apps/endatix-hub/components/survey.tsx
Comment (PeterMarinov) [inline apps/endatix-hub/components/survey.tsx:26]: This is way too common - we need to add an easy way to generate this boilerplate payload for sending a submission. I know this is what we have on the site. Just a remark, that we need to think of some utility code to help with this

File: apps/endatix-hub/package.json
Comment (PeterMarinov) [inline apps/endatix-hub/package.json:33]: Latest is great, but is dangerous as it will have different outcomes based on different time someone hit `pnpm install`. 

Dobri's PR has fixed versions. We need to use his version for this one once we resolve the conflicts. Sharing this FYI

File: apps/endatix-hub/services/api.ts
Comment (PeterMarinov) [inline apps/endatix-hub/services/api.ts:9]: TODO

- [ ] 🥘  Leftover alert. Remove please

File: src/Endatix.WebHost/appsettings.Development.json
Comment (PeterMarinov) [inline src/Endatix.WebHost/appsettings.Development.json:3]: This is intentional, right?

File: src/Endatix.WebHost/appsettings.json
Comment (PeterMarinov) [inline src/Endatix.WebHost/appsettings.json:45]: This should be empty, correct? Or maybe it's intentional. Not sure if we should keep it. Better to remove it unless I am missing something

**Review by PeterMarinov — CHANGES_REQUESTED** [review]
@orilux  - one change requested - check the TODO. Please check other comments to see what I think in terms of code evolution.

We will need to do another review before we merge to `main` the entire feature branch, so don't worry about fixing stuff unless you want to give it a try

File: apps/endatix-hub/app/(public)/share/[formId]/action.submit.ts
Comment (orilux) [reply apps/endatix-hub/app/(public)/share/[formId]/action.submit.ts:1]: OK. Done.
The reason I originally opted for `action.{action-name}.ts` is because I thought that once we have more than one action in a folder it would be organized better if they are grouped together.

File: apps/endatix-hub/app/(public)/share/[formId]/layout.tsx
Comment (orilux) [reply apps/endatix-hub/app/(public)/share/[formId]/layout.tsx:16]: Nesting inside span is valid, but apparently only for [phrasing content](https://developer.mozilla.org/en-US/docs/Web/HTML/Content_categories#phrasing_content), which doesn't include divs - I was not aware of that. I changed it to div.

File: apps/endatix-hub/app/(public)/share/[formId]/page.tsx
Comment (orilux) [reply apps/endatix-hub/app/(public)/share/[formId]/page.tsx:6]: Noted

File: apps/endatix-hub/app/(public)/share/[formId]/page.tsx
Comment (orilux) [reply apps/endatix-hub/app/(public)/share/[formId]/page.tsx:24]: I don't quite understand point 2 and point 3. If in point 3 you meant to say "move", then the three of us should coordinate first - @dspetrov is also using getFormDefinitionByFormId so we might be doing double work here and creating unnecessary conflicts.

File: apps/endatix-hub/components/survey.tsx
Comment (orilux) [reply apps/endatix-hub/components/survey.tsx:26]: Sure. Also the submission object will acquire more properties in the future, once we start doing partial submissions and meta data.

File: apps/endatix-hub/package.json
Comment (orilux) [reply apps/endatix-hub/package.json:33]: Noted.

File: apps/endatix-hub/services/api.ts
Comment (orilux) [reply apps/endatix-hub/services/api.ts]: Done

File: src/Endatix.WebHost/appsettings.Development.json
Comment (orilux) [reply src/Endatix.WebHost/appsettings.Development.json:3]: It is, but if we plan to use identical development connection strings we should coordinate on SQL Server versions and preferred SSL setup. Alternatively we could consider storing the dev connection string in a gitignored manner?

File: src/Endatix.WebHost/appsettings.json
Comment (orilux) [reply src/Endatix.WebHost/appsettings.json:45]: I am getting some weird error, so I added this gibberish key to resolve it:

```
[ 02:23:47.167 Level:ERR] An unhandled exception has occurred while executing the request.
Microsoft.Extensions.Options.OptionsValidationException: DataAnnotation validation failed for 'SendGridSettings' members: 'ApiKey' with the error: 'The ApiKey field is required.'.
```

File: src/Endatix.WebHost/appsettings.Development.json
Comment (PeterMarinov) [reply src/Endatix.WebHost/appsettings.Development.json:3]: Not a topic for now. Just a note to share in the Engineering group to discuss it. I am also using it locally.

File: src/Endatix.WebHost/appsettings.json
Comment (PeterMarinov) [reply src/Endatix.WebHost/appsettings.json:45]: This is data validation of the `SendGridSettings` - we should discuss this. For now please change it as it is misleading and offensive. It looks like a valid key, but it's not. We should rather:
1. Fix the issue with documentation and code, so devs don't have this experience
1. Add valid API Key for SendGrid - 💣 🔥 🦕  - No! No! No!
1. Another option - at least replace it with placeholder, which is honest -e.g. "{YOUR_SENDGRID_API_KEY_HERE}"

Option 2 is soo wrong. I vote for option 3. If you agree, please do it

**Review by PeterMarinov — CHANGES_REQUESTED** [review]
Only request the change for the SENDGRID_API_KEY

File: src/Endatix.WebHost/appsettings.json
Comment (orilux) [reply src/Endatix.WebHost/appsettings.json]: Option 3 is done. Thanks!

**Review by PeterMarinov — APPROVED** [review]

**Pull request merged**
